### PR TITLE
Add [GitLab] Top Language Badge

### PR DESCRIPTION
**Pull request by KoruptTinker**

### CODE REVIEW

**Comment by KoruptTinker** [comment]
Closes #10181

**Comment by KoruptTinker** [comment]
Pending: Test cases

**Comment by KoruptTinker** [comment]
Need help handling case where the API returns an empty object. Happens when the repository has no code. Only markdown? 
Any suggestions ?

**Comment by chris48s** [comment]
I'll hold off on a proper review until you've got some tests. You've said:

> Need help handling case where the API returns an empty object. Happens when the repository has no code. Only markdown?
> Any suggestions ?

Broadly there are two patterns here: One is to write the Joi schema such that this response is rejected and just accept that this case will throw the generic validation error message.

The other pattern is you can write the schema such that this response is allowed and throw an `InvalidResponse` exception if the object is empty. That allows you to customise the message, but we might need to also implement a bit of additional validation if we can't totally rely on the schema.

If you're still struggling, can you post an example of a repo or API call that has the problem you're trying to handle.

**Comment by KoruptTinker** [comment]
Test cases added. 
Ready for a review @chris48s

**Comment by KoruptTinker** [comment]
> can you post an example of a repo or API call that has the problem you're trying to handle.

https://gitlab.com/api/v4/projects/KoruptTinker%2Fgitlab-blank-repo/languages

File: services/gitlab/gitlab-top-language.service.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        message: `${languageData[topLanguage].toFixed(1)}%`,
+        color: 'blue',
+      }
+    } else {
+      return {
+        label: 'no languages found',
+        message: 'NA',
+        color: 'blue',
+      }
+    }
```
Comment (chris48s) [inline services/gitlab/gitlab-top-language.service.js:61]: Can we

```js
throw new InvalidResponse({ prettyMessage: 'no languages found' })
```

here and update the test.

File: services/gitlab/gitlab-top-language.service.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
+import Joi from 'joi'
+import { optionalUrl } from '../validators.js'
+import { pathParam, queryParam } from '../index.js'
+import GitLabBase from './gitlab-base.js'
+import { description, httpErrorsFor } from './gitlab-helper.js'
+
+const schema = Joi.object()
+  .pattern(Joi.string(), Joi.number().min(0).max(100).precision(2))
+  .required()
```
Comment (chris48s) [inline services/gitlab/gitlab-top-language.service.js:9]: The object key and value should both be `.required()`, so

```suggestion
const schema = Joi.object()
  .pattern(
    Joi.string().required(),
    Joi.number().min(0).max(100).precision(2).required(),
  )
  .required()
```

here

**Review by chris48s — COMMENTED** [review]
Thanks. Couple of comments. Aside from those, this is in good shape.

**Comment by KoruptTinker** [comment]
Requested changes made.

**Review by chris48s — APPROVED** [review]
great, thanks :+1:

**Pull request merged**